internal/cmd/byok/command_register.go

```diff
@@ -0,0 +1,222 @@
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.register,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Register a new self-managed encryption key for AWS:",
+				Code: `confluent byok register "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"`,
+			},
+			examples.Example{
+				Text: "Register a new self-managed encryption key for Azure:",
+				Code: `confluent byok register "https://a-vault.vault.azure.net/keys/a-key/00000000000000000000000000000000" --tenantid "00000000-0000-0000-0000-000000000000" --key-vault-id "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/a-resourcegroups/providers/Microsoft.KeyVault/vaults/a-vault"`,
+			},
+		),
+	}
+
+	cmd.Flags().String("key-vault-id", "", "The ID of the Azure Key Vault where the key is stored.")
+	cmd.Flags().String("tenant-id", "", "The ID of the Azure Active Directory tenant that the key vault belongs to.")
+	cmd.MarkFlagsRequiredTogether("key-vault-id", "tenant-id")
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *command) registerAWS(cmd *cobra.Command, keyString string) (*byokv1.ByokV1Key, error) {
+	keyReq := byokv1.ByokV1Key{}
+
+	keyReq.Key = &byokv1.ByokV1KeyKeyOneOf{
+		ByokV1AwsKey: &byokv1.ByokV1AwsKey{
+			KeyArn: keyString,
+			Kind:   "AwsKey",
+		},
+	}
+
+	keyResp, httpResp, err := c.V2Client.CreateByokKey(keyReq)
+	if err != nil {
+		return nil, errors.CatchCCloudV2Error(err, httpResp)
+	}
+
+	return &keyResp, nil
+}
+
+func (c *command) registerAzure(cmd *cobra.Command, keyString string) (*byokv1.ByokV1Key, error) {
+	keyReq := byokv1.ByokV1Key{}
+
+	keyVaultID, err := cmd.Flags().GetString("key-vault-id")
+	if err != nil {
+		return nil, err
+	}
+	tenantID, err := cmd.Flags().GetString("tenant-id")
+	if err != nil {
+		return nil, err
+	}
+
+	keyReq.Key = &byokv1.ByokV1KeyKeyOneOf{
+		ByokV1AzureKey: &byokv1.ByokV1AzureKey{
+			KeyId:      keyString,
+			KeyVaultId: keyVaultID,
+			TenantId:   tenantID,
+			Kind:       "AzureKey",
+		},
+	}
+
+	keyResp, httpResp, err := c.V2Client.CreateByokKey(keyReq)
+	if err != nil {
+		return nil, errors.CatchCCloudV2Error(err, httpResp)
+	}
+
+	return &keyResp, err
+
+}
+
+func (c *command) register(cmd *cobra.Command, args []string) error {
+	keyString := args[0]
+	var err error
+
+	/* outputFormat, err := cmd.Flags().GetString(output.FlagName)
+	if err != nil {
+		return err
+	} */
+
+	var postCreationStepInstructions string
+	var key *byokv1.ByokV1Key
+
+	if cmd.Flags().Changed("key-vault-id") && cmd.Flags().Changed("tenant-id") {
+		key, err = c.registerAzure(cmd, keyString)
+		if err != nil {
+			return err
+		}
+	} else if isAWSKey(keyString) {
+		key, err = c.registerAWS(cmd, keyString)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New(fmt.Sprintf("invalid key format: %s", keyString))
+	}
+
+	postCreationStepInstructions, err = renderPostCreationStepInstructions(key)
+	if err != nil {
+		return err
+	}
```

Comment:
Let's move this down to right before it's displayed? The ordering of logic is a bit weird here.

Reply:
Now: Printing only happens if no error occurs.
With change: In case that `getPostCreationStepInstructions` throws an error, we might print some information, and then an error message. I'd rather only have the error.